### PR TITLE
add prerequisites info to courses show page

### PR DESCRIPTION
**Pull request by ajm188**

### CODE REVIEW

File: app/models/course.rb [hunk 1]
```diff
@@ -5,6 +5,14 @@ class Course < ActiveRecord::Base
                                OR lower(title) like ?",
                                "%#{q.to_s.downcase.gsub(/\s+/,'')}%", "%#{q.to_s.downcase}%") }
 
+  # Get all prereq info from the database.
+  # Return an array of arrays of courses.
+  def prerequisites
+    Prerequisite.where(postrequisite: self).reduce([]) do |arr, prereq|
+      arr << Course.where('id IN (?)', prereq.prerequisite_ids)
+    end
+  end
```
Comment (Aaronneyer) [inline app/models/course.rb]: Is there no way to do this with Rails associations?

File: app/views/courses/show.html.erb [hunk 2]
```diff
@@ -57,7 +57,18 @@
...
+                  <p>
+                  <ul>
+                    <% prereq_sets = @course.prerequisites %>
+                    <% if prereq_sets.empty? %>
+                      <li>None</li>
+                    <% else %>
+                      <% @course.prerequisites.each do |prereq_set| %>
+                        <li><%= prereq_set.map(&:to_s).join(" or ") %></li>
+                      <% end%>
+                    <% end %>
```
Comment (Aaronneyer) [inline app/views/courses/show.html.erb]: Seems like too much logic for the view layer, perhaps move most of this to a helper method?

i.e. something like this:
```
def prereq_sets(course)
  if prereq_sets.empty?
    concat content_tag(:li, "None")
  else
    course.prerequisites.each do |prereq_set|
      concat content_tag(:li, prereq_set.map(&:to_s).join(" or "))
    end
  end
end
```

File: app/models/course.rb (hunk 1 above)
Comment (ajm188) [inline app/models/course.rb]: I'm not sure if that will work with it being an array in the database...I'll look into it though

Comment (ajm188) [inline app/models/course.rb]: http://stackoverflow.com/questions/17012330/rails-postgres-foreign-keys-stored-in-array-to-create-1-many-association

File: app/views/courses/show.html.erb [hunk 3]
```diff
@@ -96,7 +102,15 @@
...
-<p>Prereqs: TODO</p>
+<p>Prereqs: 
+<ul>
+  <% @course.prerequisites.each do |prereq_set| %>
+    <li>
+    <%= prereq_set.map(&:long_string).join(" or ") %>
+    </li>
+  <% end %>
+</ul>
+</p>
```
Comment (Aaronneyer) [inline app/views/courses/show.html.erb]: Any reason for all the commented out code down here?

Aside from this, looks good. :+1:

**Comment by ajm188** [comment]
I don't even remember doing that lol. Maybe it was for the demo? Honestly not sure

**Pull request merged**